pkg/detach/copy.go

```diff
@@ -11,47 +12,73 @@ var ErrDetach = errors.New("detached from container")
 
 // Copy is similar to io.Copy but support a detach key sequence to break out.
 func Copy(dst io.Writer, src io.Reader, keys []byte) (written int64, err error) {
+	// if no key sequence we can use the fast std lib implementation
+	if len(keys) == 0 {
+		return io.Copy(dst, src)
+	}
 	buf := make([]byte, 32*1024)
+	tmpKeyBuf := make([]byte, 0, len(keys))
+outer:
 	for {
 		nr, er := src.Read(buf)
-		if nr > 0 {
-			preservBuf := []byte{}
-			for i, key := range keys {
-				preservBuf = append(preservBuf, buf[0:nr]...)
-				if nr != 1 || buf[0] != key {
-					break
-				}
-				if i == len(keys)-1 {
-					return 0, ErrDetach
+
+		// previous key buffer
+		if len(tmpKeyBuf) > 0 {
```

Comment:
Would it be faster to just do a linear search of buf instead of copying into a separate, smaller buffer?

Reply:
Do you mean like store the index of where we are in the middle of the sequence before the next read is done? That sound like good plan. I was to much into we must keep writing the "buffered" sequence when a later key does not match but in that case we can get the correct bytes just out of the input keys slice instead instead of the tmpKeyBuf.

Reply:
If you start searching from (index in buffer) - (length of detach sequence) + 1, you should be able to catch the full sequence even if only one character of the sequence was in the most recent block.